.github/workflows/test.yml

```diff
@@ -0,0 +1,146 @@
+# This is a GitHub workflow defining a set of jobs with a set of steps.
+# ref: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions
+#
+name: Test
+
+on:
+  pull_request:
+    branches: ["**"]
+  push:
+    branches: [main, master]
+    tags: ["**"]
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  test_jupyterhub:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 10
+
+    env:
+      # NOTE: UTF-8 content may be interpreted as ascii and causes errors
+      #       without this.
+      LANG: C.UTF-8
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - python: "3.6"
+            pre_commit: true
```

Comment:
I think it's clearer to move pre-commit to a separate lint job instead of including it in the matrix. The required steps are fairly trivial so I don't think the added complexity of the if conditionals is worth it, and logically it's different to the oher conditionals which affect the test scenario.
```
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-python@v2
    - uses: pre-commit/action@v2.0.0
```
In addition it's easier for developers to see the lint error show up immediately in it's own job- it'll often show up in a few seconds since there's not much setup involved.

Reply:
Excellent point, added as a separate step, there was very little duplication as we didn't need to install anything and what you listed above works to run pre-commit!